playbooks/roles/mysql/files/edxapp.sql

```diff
@@ -31,5 +31,112 @@ VALUES
   NOW(),
   True,
   '{"default":{"platform_name":"Microsoft","company_about_url":"https://www.microsoft.com/en-us/about","company_privacy_url":"https://privacy.microsoft.com/en-us/privacystatement/","company_tos_url":"https://openedx.microsoft.com/tos","logo_src":"https://openedx.microsoft.com/static/themes/default/images/ms-logo.png","logo_url":"http://www.microsoft.com"},"honor":{"certificate_type":"honor","certificate_title":"Honor Certificate","document_body_class_append":"is-honorcode"},"verified":{"certificate_type":"verified","certificate_title":"Verified Certificate","document_body_class_append":"is-idverified"},"base":{"certificate_type":"base","certificate_title":"Certificate of Achievement","document_body_class_append":"is-base"},"distinguished":{"certificate_type":"distinguished","certificate_title":"Distinguished Certificate of Achievement","document_body_class_append":"is-distinguished"}}'
- );
+);
+
+/*
+  Insert the entry so that self_paced courses are allowed on the platform.
+
+  In the self_paced_selfpacedconfiguration table we can have only one row defined so first delete any existing ones
+  and then insert the selfpacedconfiguration. It is inserted as enabled, ready to be used.
+  
+  It is safe to run this multiple times.
+*/
+
+DELETE FROM self_paced_selfpacedconfiguration;
+
+INSERT INTO self_paced_selfpacedconfiguration 
+(
+  change_date,
+  enabled,
+  enable_course_home_improvements
+) 
+VALUES
+(
+  NOW(),
+  1,
+  1
+);
+
+
+/*
+  Insert the entry so that certificate generation is allowed on the platform.
+
+  In the certificates_certificategenerationconfiguration table we can have only one row defined so first delete any existing ones
+  and then insert the certificategenerationconfiguration. It is inserted as enabled, ready to be used.
+
+  It is safe to run this multiple times.
+*/
+DELETE FROM certificates_certificategenerationconfiguration;
+
+INSERT INTO certificates_certificategenerationconfiguration 
+(
+  change_date,
+  enabled
+) 
+VALUES
+(
+  NOW(),
+  1
+);
+
+
+/*
+  Create the super user that will be used for DRI. We will use oxamaster@microsoft.com for this account.
+  We must have auth_user and auth_userprofile entries for oxamaster. We first delete these rows if they exist in the tables.
+  So this code is rerunnable. 
+  
+  If we create courses or enroll for courses in production environment there will be FK entries in other tables that will block us from deleting these entries if we rerun. 
+  In such a case we can use django admin panel to delete superuser if needed. Later we can add logic here don't try to recreate if it already exists. 
+
+  We will set is_active = False. So we must use reset password inorder to set password and to activate the account.
+*/
+DELETE FROM auth_userprofile where user_id in (select id FROM auth_user WHERE username='oxamaster');
+DELETE FROM auth_user WHERE username='oxamaster';
```

Comment:
I don't think the delete and insert approach is going to work. It doesn't account for relations to other tables - for instance, if the superuser has made progress on any course.

I think we should look for an approach that does insert if not exist.